pkg/certificate/providers/config.go

```diff
@@ -208,11 +196,8 @@ func (c *MRCProviderGenerator) getHashiVaultOSMCertificateManager(mrc *v1alpha2.
 	if err != nil {
 		return nil, "", fmt.Errorf("error instantiating Hashicorp Vault as a Certificate Manager: %w", err)
 	}
-	id, err := getMRCID(mrc)
-	if err != nil {
-		return nil, "", err
-	}
-	return vaultClient, id, nil
+
+	return vaultClient, mrc.Name, nil
```

Comment:
At this point, i would change the func signature to (issuer, error), and let the manager set the ID on it's own

Reply:
Updated the signature to no longer return the issuer id.